server/bigbluebuttonapiwrapper/api/APIcalls.go

```diff
@@ -0,0 +1,321 @@
+// if there are no attendees currently present in the meeting
+
+// see http://docs.bigbluebutton.org/dev/api.html for API documentation
+func CreateMeeting(meetingRoom *dataStructs.MeetingRoom) string {
+	if meetingRoom.Name_ == "" || meetingRoom.MeetingID_ == "" ||
+		meetingRoom.AttendeePW_ == "" || meetingRoom.ModeratorPW_ == "" {
+		log.Println("ERROR: PARAM ERROR.")
+		return "ERROR: PARAM ERROR."
+	}
+
+	name := "name=" + url.QueryEscape(meetingRoom.Name_)
+	meetingID := "&meetingID=" + url.QueryEscape(meetingRoom.MeetingID_)
+	attendeePW := "&attendeePW=" + url.QueryEscape(meetingRoom.AttendeePW_)
+	moderatorPW := "&moderatorPW=" + url.QueryEscape(meetingRoom.ModeratorPW_)
+	welcome := "&welcome=" + url.QueryEscape(meetingRoom.Welcome)
+	dialNumber := "&dialNumber=" + url.QueryEscape(meetingRoom.DialNumber)
+	logoutURL := "&logoutURL=" + url.QueryEscape(meetingRoom.LogoutURL)
+	record := "&record=" + url.QueryEscape(meetingRoom.Record)
+	duration := "&duration=" + url.QueryEscape(strconv.Itoa(meetingRoom.Duration))
+	allowStartStopRecording := "&allowStartStopRecording=" +
+		url.QueryEscape(strconv.FormatBool(meetingRoom.AllowStartStopRecording))
+	moderatorOnlyMessage := "&moderatorOnlyMessage=" +
+		url.QueryEscape(meetingRoom.ModeratorOnlyMessage)
+	meta_bn_recording_ready_url := "&meta_bn-recording-ready-url=" +
+		url.QueryEscape(meetingRoom.Meta_bn_recording_ready_url)
+	meta_channelid := "&meta_channelid=" +
+		url.QueryEscape(meetingRoom.Meta_channelid)
+	meta_endcallback := "&meta_endcallbackurl=" +
+		url.QueryEscape(meetingRoom.Meta_endcallbackurl)
+	voiceBridge := "&voiceBridge=" + url.QueryEscape(meetingRoom.VoiceBridge)
+
+	createParam := name + meetingID + attendeePW + moderatorPW + welcome + dialNumber +
+		voiceBridge + logoutURL + record + duration + moderatorOnlyMessage + meta_bn_recording_ready_url + meta_channelid +
+		meta_endcallback + allowStartStopRecording
+
+	checksum := helpers.GetChecksum("create" + createParam + SALT)
+
+	response := helpers.HttpGet(BaseUrl + "create?" + createParam + "&checksum=" +
+		checksum)
+
+	if "ERROR" == response {
+		log.Println("ERROR: HTTP ERROR.")
+		return "ERROR: HTTP ERROR."
+	}
+	err := helpers.ReadXML(response, &meetingRoom.CreateMeetingResponse)
+
+	if nil != err {
+		log.Println("XML PARSE ERROR: " + err.Error())
+		return "ERROR: XML PARSE ERROR."
+	}
+
+	if "SUCCESS" == meetingRoom.CreateMeetingResponse.Returncode {
+		log.Println("SUCCESS CREATE MEETINGROOM. MEETING ID: " +
+			meetingRoom.CreateMeetingResponse.MeetingID)
+		return meetingRoom.CreateMeetingResponse.MeetingID
+	} else {
+		log.Println("CREATE MEETINGROOM FAILD: " + response)
+		return "FAILED"
+	}
+	return "ERROR: UNKNOWN."
+}
+
+// GetJoinURL: we send in a Participant struct and get back a joinurl that participant can go to
+func GetJoinURL(participants *(dataStructs.Participants)) string {
+	if "" == participants.FullName_ || "" == participants.MeetingID_ ||
```

Comment:
Maybe move this into a `isValid()` method?

Reply:
Done.